dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/controller/impl/BaseStepControllerImpl.java

```diff
@@ -437,7 +437,8 @@ protected void markStepForFailure() {
     // Will be overridden for partitioned steps by special method which
     // aggregates partition-level metrics
     protected void updateStepExecution() {
-        getPersistenceManagerService().updateStepExecution(runtimeStepExecution);
+        StepThreadExecutionEntity stepThreadExecutionEntity = getPersistenceManagerService().updateStepExecution(runtimeStepExecution);
+        stepThreadExecutionEntity.getJobExecution().setLastUpdatedTime(runtimeStepExecution.getLastUpdatedTime());
     }
```

Comment:
Seems like we should still mark job lastUpdated for partition threads  (since the previous setLastUpdatedTime marked it for each.  But this comment seems to say it won't be..

Reply:
You're right here in that I missed the override for the partition case and I'll make that revision.  But even in that case, it looks like it is only marking the lastUpdated for the top level (see updateStepExecutionWithPartitionAggregate in JPAPersistenceManagerImpl)  The only thing it does with the partitions is to aggregate the metrics (which the comment indicates)

Reply:
Hmm... not following your comment 100% but it made me see a mistake in my comment.   I was wrong to say your code doesn't update lastUpdated for partition _threads_... the missing gap is the top-level thread in partitioned steps.  That's the only gap that needs covering.

The `BaseStepControllerImpl.updateStepExecution()` method here runs on the partitioned threads (just not on the top-level thread of a partitioned step).. I believe.

Are we on the same page now?

Reply:
I was referring to PartitionedStepControllerImpl.updateStepExecution() which I had missed.  It would need the same treatment as BaseStepControllerImpl. (They both use the same AbstractPersistenceManager.updateStepExecutionStatusTimeStampsUserDataAndMetrics where the set lastUpdated was moved out of)